### DISCUSSION TITLE
Making particles follow a line and then disapear

### DISCUSSION THREAD

**Discussion by gozdecoban** [matteobruni/tsparticles#1288] [Q&A] [answered]
I am getting lost in the options, I am trying to do something like a shooting star with star particles follow the path. It should start in a certain location and end where my star component will appear. The first created particles will fade after some time. How can I manage these particles to follow a path? Also how can I trigger the beginning of the creation of the particles based on some event?

Edit: I am using React with typescript

**Comment by matteobruni** [comment, marked as answer]
Hi @egozde,

I don't know if you can achieve everything with the current React component but I will try to help you.

For particles disappearing after some time you can see this sample: https://codepen.io/matteobruni/pen/ZEWbYzj using the `life` options

For customizing particles path you can see this sample: https://codepen.io/matteobruni/pen/WNrPOYE using the `setNoise` method of the container. It's not necessary all the code you see in the sample, what you need is the `setNoise` function that has some function arguments to alter the particle speed or position when moving

About particles creation you can use the container method `addParticle` like in this sample: https://codepen.io/matteobruni/pen/GRpYdGv and like in the `setNoise` sample you just need to see how to use `addParticle`, if you don't specify the position the particle will be randomly positioned

There's a lot to explore I know, I'm here for any further help.

It looks everything compatible with the React component but I'm not 100% sure, if it's not possible you need to use only `tsparticles` without `react-tsparticles` with your custom component

**Reply by gozdecoban** [reply]
@matteobit Thanks for your answer! Is it possible to trigger a click event from outside? Also is it possible to make particles appear somewhere else than the edge of the cursor when clicked?

**Reply by matteobruni** [reply]
You can add a `click` listener to the canvas and execute your custom code. `addParticle` without arguments adds a Particle to the canvas following the global options at a random position.

There’s also the `setOnClickHandler`, you can find it in the Readme
